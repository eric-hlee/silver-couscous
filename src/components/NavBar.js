import React from 'react';
import Navbar from 'react-bootstrap/Navbar';
import { FormattedMessage } from 'react-intl';

export default function NavBar() {
	return (
    <Navbar bg="light" expand="lg">
      <Navbar.Brand href="/">
        <FormattedMessage id="navbar.text" defaultMessage="Eric Lee" />
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
    </Navbar>
	)
}
