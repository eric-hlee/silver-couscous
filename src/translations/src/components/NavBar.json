[{"id":"navbar.text","defaultMessage":"Eric Lee","file":"src/components/NavBar.js","start":{"line":9,"column":8},"end":{"line":9,"column":71}}]
