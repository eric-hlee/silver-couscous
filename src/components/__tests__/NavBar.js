import React from 'react'
import { IntlProvider } from 'react-intl'
import { shallow, mount } from 'enzyme'
import NavBar from '../NavBar'

describe('<NavBar />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<NavBar />)
    expect(wrapper).toMatchSnapshot()
  })

  it('switches between languages when toggle button clicked', () => {
    const onChangeLanguage = jest.fn()
    const wrapper = mount(
      <IntlProvider locale="en">
        <NavBar onChangeLanguage={onChangeLanguage} />
      </IntlProvider>
    )
    wrapper.find('Button').simulate('click')
    expect(onChangeLanguage).toHaveBeenCalled()
  })
})
