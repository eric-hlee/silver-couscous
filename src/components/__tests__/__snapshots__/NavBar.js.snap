// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders correctly 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  variant="light"
>
  <NavbarBrand
    href="/"
  >
    <FormattedMessage
      defaultMessage="Eric Lee"
      id="navbar.text"
      values={Object {}}
    />
  </NavbarBrand>
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    label="Toggle navigation"
  />
</Navbar>
`;
